### PR TITLE
CO-3911 Don't display the segmentation survey on MyCompassion for W&P

### PR DESCRIPTION
**Pull request by TheRaphael0000**
- add a computed field to know if the partner participate to the W&P program
- if the partner participate to the W&P program, ignore the survey

### CODE REVIEW

File: partner_compassion/models/partner_compassion.py [hunk 1]
```diff
@@ -255,6 +255,13 @@ class ResPartner(models.Model):
         track_visibility="onchange"
     )
 
+    write_and_pray = fields.Boolean(
+        string="Write & Pray",
+        help="Have at least one sponsorship for the W&P program",
+        compute="_compute_write_and_pray",
+        store=False,
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]:
```suggestion
```

File: partner_compassion/models/partner_compassion.py [hunk 2]
```diff
@@ -318,6 +325,12 @@ def _compute_criminal_record_name(self):
                     partner.with_context(bin_size=False).criminal_record)
                 partner.criminal_record_name = f"Criminal record {partner.name}{ftype}"
 
+    @api.multi
+    @api.depends("sponsorship_ids")
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]: Useless for non-stored field.
```suggestion
```

File: partner_compassion/models/partner_compassion.py [hunk 3]
```diff
@@ -318,6 +325,12 @@ def _compute_criminal_record_name(self):
...
+    def _compute_write_and_pray(self):
+        for partner in self:
+            partner.write_and_pray = any([t == "SWP" for t in partner.mapped("sponsorship_ids.type")])
```
Comment (ecino) [inline partner_compassion/models/partner_compassion.py]:
```suggestion
            partner.write_and_pray = "SWP" in partner.mapped("sponsorship_ids.type")
```

**Review by ecino — CHANGES_REQUESTED** [review]

**Comment by ecino** [comment]
I changed the base branch as this can be put in production earlier.

File: partner_compassion/models/partner_compassion.py (hunk 3 above)
Comment (TheRaphael0000) [reply partner_compassion/models/partner_compassion.py]: Thank you, I knew this could be simplified

**Review by ecino — APPROVED** [review]

**Pull request merged**